[Siyan] iP

Siyan-G opened the pull request:
# Chatty 🗨️  
> “Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort.” – Paul J. Meyer  

Meet **Chatty**, your personal task management assistant! Unlike traditional to-do apps, Chatty interacts with you just like a friendly assistant, helping you keep track of your tasks with ease. It's:  

- **Conversational** – Manage tasks naturally through chat 💬  
- _Easy to use_ – No complicated commands needed! 🎓  
- ~~Slow~~ **Lightning fast** – No more forgetting deadlines ⚡  

## Getting Started 🚀  
Using Chatty is as simple as:  

1. **Download** it from [here](#).  
2. **Run it** with a double-click.  
3. **Chat with Chatty** to manage your tasks.  
4. Stay on top of your schedule effortlessly! 😉  

## Features ✨  
- [x] Create and track **todos**  
- [x] Schedule and manage **events**  
- [x] Set **deadlines** with ease  

## For Developers 👨‍💻👩‍💻  
Chatty is built in Java, making it a great project for Java developers to explore object-oriented programming. Here’s a look at the `main` method:  

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Siyan-G committed:
rename file name, add intro and exit messages

Siyan-G committed:
modified intro and exit messages

Siyan-G committed:
Echo

Siyan-G committed:
Add, List

Siyan-G committed:
Mark as done

Siyan-G committed:
ToDo, Event, Deadline

Siyan-G committed:
Automated Text UI Testing

Siyan-G committed:
Handle Errors

Siyan-G committed:
Delete

kumar2215 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,176 @@
+
+public class Chatty {
+    private static final String linebreak = "___________________________________________________________________";
+    private static final String introMsg = "\n\tHello Master! I'm Chatty, your ever-ready personal assistant.\n\tHow can I help you today?";
```
I like how you created some constants here for some of the bots' messages. However, note that according to the Java Coding standard in this course, constants use the screaming snake case instead of the camel case used for normal variables.

Siyan-G replied:
thanks for the input, have made the relevant changes!

kumar2215 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,176 @@
+    private static final String introMsg = "\n\tHello Master! I'm Chatty, your ever-ready personal assistant.\n\tHow can I help you today?";
+    private static final String exitMsg = "\n\tGoodbye Master! See you soon!";
+    private ArrayList<Task> taskList = new ArrayList<>();
+    private Integer taskCounter = 0;
```
Any reason why you used the Wrapper class `Integer` instead of the primitive type `int`?

Siyan-G replied:
thanks for the input, have changed it to primitive type

kumar2215 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,176 @@
+    public void handleCommand(String command) {
+        try{
+            // handle list command
+            if (command.equalsIgnoreCase("list")) {
```
The other commands do not allow for case insensitivity. So why is the list command special? Maybe consider changing command to a single case (lower or upper) first before processing it if you want to allow for case insensitivity that is.

Siyan-G replied:
thanks for the input, have made relevant changes to make it consistent

kumar2215 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,176 @@
+    public void deleteTask(String command) throws ChattyTaskNotFoundException, ChattyInvalidArgumentException {
+        String[] commandParts = command.split(" ");
+        if (commandParts.length == 2) {
+            int taskId = Integer.parseInt(commandParts[1].trim());
```
Perhaps consider the case where the user types "delete study" where they didn't think of using the index at first. This would cause an `NumberFormatException` when trying to parse the 2nd word as a number which isn't handled.

Siyan-G replied:
thanks for the input, it is really helpful, will make relevant changes

kumar2215 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,176 @@
+    public void setTaskStatus(String command) throws ChattyTaskNotFoundException {
+        String[] commandParts = command.split(" ");
+        if (commandParts.length == 2) {
+            int taskId = Integer.parseInt(commandParts[1].trim());
```
Same issue as the one mentioned for `deleteTask`. Perhaps consider placing the parsing of the index number in its own method since its being used in more than 1 place.

Siyan-G replied:
thanks for the input, it is really helpful, will make relevant changes

kumar2215 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,176 @@
+        }
+    }
+
+    public void handleTaskCreation(String command) throws ChattyInvalidArgumentException {
```
Perhaps the processing of the description string for each task can be handled in the respective Task subclasses instead?

Siyan-G replied:
thanks for the input, will make relevant changes

kumar2215 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,176 @@
+        }
+        if (command.startsWith("event")) {
+            String taskString = command.substring(5).trim();
+            String[] taskParts = taskString.split("/");
```
What is the allowed format for event tasks here? If it is "name /from start /to end", the strings "from" and "to" are not checked, meaning the string "name /start start /end end" might be incorrectly parsed.

Siyan-G replied:
thanks for the input, will make relevant changes

kumar2215 commented on src/main/java/exception/ChattyInvalidArgumentException.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String getMessage() {
+        return String.format(" Invalid argument detected for command: %s", commandType);
```
Maybe informing the user of what the correct format should be can be useful.

Siyan-G replied:
thanks for the input, will make relevant changes

kumar2215 commented on src/main/java/Chatty.java:
A lot of the methods used in the class seem to be called internally in this file alone. Perhaps consider making those that are not going to be called outside this class to be made private?

Siyan-G replied:
thanks for the input, will make relevant changes

kumar2215 reviewed (commented):
Here are some issues that came to my mind. Certain issues might be a bit nitpicking while the rest are more concerned with user functionality and something you might want to consider looking into.

Siyan-G committed:
Save

Siyan-G committed:
Merge branch 'branch-Level-7'

Siyan-G committed:
Dates and Times

Siyan-G committed:
Merge branch 'branch-Level-8'

themintchoco commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,193 @@
+        this.taskCounter = taskCounter;
+    }
+
+    private void Response(String message) {
+        System.out.println("Chatty:" + message);
+        System.out.println(LINE_BREAK);
+    }
+
+    public void Intro() {
+        System.out.println(LINE_BREAK);
+        this.Response(INTRO_MSG);
+    }
+
+    public void Exit() {
+        this.Response(EXIT_MSG);
+    }
```
Method names should be in camelCase

Siyan-G replied:
thanks for the input, will make relevant changes

themintchoco commented on src/main/java/Deadlines.java:
It may be better to name this class `Deadline` (singular form), since it represents a single deadline.

Siyan-G replied:
thanks for the input, have made the relevant changes

themintchoco commented on src/main/java/Events.java:
Similar to `Deadlines`, it may be better to name this class `Event` (singular form), since it represents a single event.

Siyan-G replied:
thanks for the input, have made the relevant changes

themintchoco reviewed (commented):
Thanks for the PR! Just a few minor nits. Great work!

Siyan-G committed:
restructure for OOP

Siyan-G committed:
organise into packages

Siyan-G committed:
Merge branch 'add-gradle-support'

Siyan-G committed:
add junit test class for Todo and DeleteCommand classes

Siyan-G committed:
set up jar environment

Siyan-G committed:
add javadoc to classes

Siyan-G committed:
tweak coding standard

Siyan-G committed:
Merge branch 'branch-A-CodingStandard'

Siyan-G committed:
merge branch-A-Javadoc into main

Siyan-G committed:
add find command

Siyan-G committed:
merge branch-Level-9 into main

Siyan-G committed:
add imports

Siyan-G committed:
add checkstyle

Siyan-G committed:
Merge branch 'branch-A-CheckStyle'

Siyan-G committed:
add javafx

Siyan-G committed:
Merge branch 'branch-Level-10'

Siyan-G committed:
Add assertions to various classees

Siyan-G committed:
Fix tasksContain method in TaskList class

Siyan-G committed:
Parser class: extract individual methods for parsing commands

The parse method is over 50 lines long, handling multiple commands in a single block of code. This makes it difficult to maintain, especially as new commands are added.

Each command has a unique format and requires different parsing logic to extract relevant details before execution. Keeping all logic in a single method leads to poor readability and maintainability.

Extract individual static methods for parsing each command.

By breaking the parse method into smaller, dedicated parsing methods, we improve readability, reduce code duplication, and make it easier to extend the parser in the future.

Using separate methods allows for better error handling and validation at a more granular level, making debugging and future modifications more manageable.

Siyan-G committed:
Refactor TaskList to use Java Streams

The TaskList class currently uses traditional loops and manual list operations, making the code less concise and harder to read.

Refactor TaskList to leverage Java Streams for operations such as filtering, mapping, and collecting. Remove redundant numOfTasks field and replace manual iterations with stream-based processing.

Using Streams improves code readability, reduces boilerplate, and aligns with modern Java best practices. The refactored methods are more expressive and maintainable while preserving existing functionality.

All Javadoc comments are retained, ensuring documentation remains up-to-date.

Siyan-G committed:
Merge pull request #1 from Siyan-G/branch-A-Assertions

Add assertions to verify program state at key execution points

Siyan-G committed:
Merge branch 'master' into branch-A-CodeQuality

Siyan-G committed:
Merge pull request #2 from Siyan-G/branch-A-CodeQuality

Parser class: extract individual methods for parsing commands

Siyan-G committed:
Revert "Parser class: extract individual methods for parsing commands"

Siyan-G committed:
Merge pull request #3 from Siyan-G/revert-2-branch-A-CodeQuality

Revert "Parser class: extract individual methods for parsing commands"

Siyan-G committed:
Parser class: extract individual methods for parsing commands

The parse method is over 50 lines long, handling multiple commands in a single block of code. This makes it difficult to maintain, especially as new commands are added.

Each command has a unique format and requires different parsing logic to extract relevant details before execution. Keeping all logic in a single method leads to poor readability and maintainability.

Extract individual static methods for parsing each command.

By breaking the parse method into smaller, dedicated parsing methods, we improve readability, reduce code duplication, and make it easier to extend the parser in the future.

Using separate methods allows for better error handling and validation at a more granular level, making debugging and future modifications more manageable.

Siyan-G committed:
conclude merge

Siyan-G committed:
Merge branch 'master' into branch-A-Streams

Siyan-G committed:
Merge pull request #4 from Siyan-G/branch-A-Streams

Refactor TaskList to use Java Streams

Siyan-G committed:
Edit message format for 'find' and 'list' commands.

The message format for the 'find' and 'list' commands use wrong line breaks.

Update the message format for both commands to be more consistent and each task is printed on a new line.

These changes improve the clarity of command output and improves user experience.

Siyan-G committed:
minor refract of code in TaskList and FindCommand classes

Siyan-G committed:
Create chatty.yml

Set up CI workflow

Siyan-G committed:
Merge pull request #5 from Siyan-G/branch-A-CI

Set up Java CI workflow

Siyan-G committed:
add help command to display list of commands available for use

Siyan-G committed:
edit the help message in the UI class

Siyan-G committed:
add javadoc in HelpCommand class

Siyan-G committed:
update help message in Ui class

Siyan-G committed:
Merge branch 'branch-A-CodeQuality'

Siyan-G committed:
break down parse method into methods for individual commands

Siyan-G committed:
add parsing of help command in Parse method

Siyan-G committed:
Merge branch 'branch-B-Help'

Siyan-G committed:
Edit app title and logo for GUI

Siyan-G committed:
Adjust the size of icon in DialogBox to be smaller

Siyan-G committed:
Set window to be resizeable

Siyan-G committed:
Edit intro message and help message

Siyan-G committed:
Change icon to circular

Siyan-G committed:
Merge branch 'branch-A-BetterGUI'

Siyan-G committed:
Edit message format to give chatty a personality

Siyan-G committed:
Merge branch 'branch-A-Personality'

Siyan-G committed:
Use Enum to identify command type in ChattyInvalidCommandFormatException class

Siyan-G committed:
Amend error handling for invalid command, invalid command arguments

Siyan-G committed:
Edit taskId attribute in ChattyTaskNotFoundException class to be final

Siyan-G committed:
Amend error message format and parsing of Event command when format is incorrect

Siyan-G committed:
Merge branch-A-MoreErrorHandling into master

Siyan-G committed:
Amend the welcome message

Siyan-G committed:
Fix coding standard at Launcher class main method

Siyan-G committed:
Edit error message in ChattyInvalidCommand class

Siyan-G committed:
Add Ui png for user guide

Siyan-G committed:
Amend readme for user guide

Siyan-G committed:
Remove duplicated lines in readme

Siyan-G committed:
Update chatty.yml

attempt to fix failed run

Siyan-G committed:
Amend readme for user guide

Siyan-G committed:
Merge branch 'master' of https://github.com/Siyan-G/ip

Siyan-G committed:
Edit action line 7 in readme for user guide

Siyan-G committed:
Amend event command details in readme for user guide

Siyan-G committed:
Add missing periods in command return messages.

Siyan-G committed:
Fix typo in UnmarkCommand class

Siyan-G committed:
Add placeholder temporarily for links in readme.

Siyan-G committed:
Add Gui screenshot into readme

Siyan-G committed:
Add java installation guide link for mac in readme.

Siyan-G committed:
Amend DeleteCommandTest class

Siyan-G committed:
Amend code block to run application in readme

Siyan-G committed:
Fix typo in introduction message

Siyan-G committed:
Update readme for user guide

Siyan-G committed:
Remove extra blank line in Storage class

Siyan-G committed:
Update readme file to fix format in github page
